spec/features/home_index_spec.rb

```diff
@@ -18,4 +30,117 @@
+      scenario '正しく値を入力した場合、検索結果があること' do
+        visit root_path
+        select 'イヌ', from: 'q_category_eq'
+        select 'オス', from: 'q_gender_eq'
+        fill_in 'q_age_lteq', with: '12'
+        select 'チワワ', from: 'q_classification_eq'
+        check '大阪'
+        click_button '検索する'
+        expect(page).to have_content 'taro20221101'
+      end
+    end
+
+    context '個々検索' do
+      context 'カテゴリ' do
+        scenario 'カテゴリでの登録がある場合、検索結果があること' do
+          visit root_path
+          select 'イヌ', from: 'q_category_eq'
+          click_button '検索する'
+          expect(page).to have_content 'taro20221101'
+        end
+        scenario 'カテゴリでの登録がない場合、検索結果がないこと' do
+          visit root_path
+          select 'ネコ', from: 'q_category_eq'
+          click_button '検索する'
+          expect(page).not_to have_content 'taro20221101'
+        end
+      end
+
+      context '性別' do
+        scenario '性別での登録がある場合、検索結果があること' do
+          visit root_path
+          select 'オス', from: 'q_gender_eq'
+          click_button '検索する'
+          expect(page).to have_content 'taro20221101'
+        end
+        scenario '性別での登録がない場合、検索結果がないこと' do
+          visit root_path
+          select 'メス', from: 'q_gender_eq'
+          click_button '検索する'
+          expect(page).not_to have_content 'taro20221101'
+        end
+      end
+      context '年齢' do
+        scenario '年齢の登録範囲内（より若い）の場合、検索結果があること' do
+          visit root_path
+          fill_in 'q_age_lteq', with: '12'
+          click_button '検索する'
+          expect(page).to have_content 'taro20221101'
+        end
+        scenario '年齢の登録範囲内（より若い）でない場合、検索結果がないこと' do
+          visit root_path
+          fill_in 'q_age_lteq', with: '11'
+          click_button '検索する'
+          expect(page).not_to have_content 'taro20221101'
+        end
+      end
+      context '種別' do
+        scenario '種別での登録がある場合、検索結果があること' do
+          visit root_path
+          select 'チワワ', from: 'q_classification_eq'
+          click_button '検索する'
+          expect(page).to have_content 'taro20221101'
+        end
+        scenario '種別での登録がない場合、検索結果がないこと' do
+          visit root_path
+          select 'ダックス', from: 'q_classification_eq'
+          click_button '検索する'
+          expect(page).not_to have_content 'taro20221101'
+        end
+      end
+
+      context '譲渡可能地域' do
+        scenario '譲渡可能地域での登録がある場合、検索結果があること' do
+          visit root_path
+          check '大阪'
+          click_button '検索する'
+          expect(page).to have_content 'taro20221101'
+        end
+        scenario '譲渡可能地域での登録がない場合、検索結果がないこと' do
+          visit root_path
+          check '東京'
+          click_button '検索する'
+          expect(page).not_to have_content 'taro20221101'
+        end
+      end
+
+      context '複数譲渡可能地域登録検索' do
+        before do
+          # TODO この部分ですが、もっと綺麗な書き方があると思うが、今はこれで。
+          @area = create(:area, place_name: '和歌山')
+          @pet_area = create(:pet_area, pet_id: @pet.id, area_id: @area.reload.id)
```

Comment:
ブロックを活用すれば↓のようにも書けます。

```suggestion
          create(:area, place_name: '和歌山') do |area|
              create(:pet_area, pet_id: @pet.id, area_id: area.id)
          end
```

あと特に使われないものは変数に入れなくて OK です。